uninstall/uninstall_linux.go

```diff
@@ -0,0 +1,90 @@
+package uninstall
+
+import (
+	"time"
+	"io/ioutil"
+	"os/exec"
+	"github.com/pufferpanel/pufferd/logging"
+	"syscall"
+)
+
+
+func StartProcess(){
+	killDaemon()
+	deleteFiles()
+	deleteUser()
+}
+
+func killDaemon(){
+	exec.Command("killall", "-15" "-u", "pufferd").Run()
+	logging.Info("Attempting to kill all pufferd process...")
+	time.Sleep(time.Second * 5)//Giving 5 seconds to kill "correctly" all process
+	exec.Command("killall", "-9" "-u", "pufferd").Run()//"Hard killing" anything
+}
+
+func deleteUser(){
+	cmd := exec.Command("userdel", "-Z", "-r", "-f", "pufferd")
+	err := cmd.Run() //Delete pufferd and it's home dir (/var/lib/pufferd)
+
+	if err != nil{
+
+		if exitErr, ok := err.(*exec.ExitError); ok {
+
+			if status, ok := exitErr.Sys().(syscall.WaitStatus); ok {
+				switch status.ExitStatus(){
+					case 6:
+						logging.Error("The pufferd user don't exist", err)
+					case 8:
+						logging.Error("The pufferd user is logged in", err)
+					case 12://Wrong way, I know
+						logging.Error("	Couldn't remove pufferd directory", err)
+					case 10://Wrong way, I know
+						logging.Error("Couldn't update group file", err)
+					
+				}
+			}
+		}
+		logging.Error("Couldn't delete the pufferd user")
+	}
+
+
+
+
+	cmd = exec.Command("groupdel", "pufferd")

```

Comment:
None of us were wrong actually :P 
Linux should delete a group once it's empty...
```
root@ip-xxx-xxx-xxx-xxx:~# adduser apollonio
Adding user `apollonio' ...
Adding new group `apollonio' (1001) ...
Adding new user `apollonio' (1001) with group `apollonio' ...
The home directory `/home/apollonio' already exists.  Not copying from `/etc/skel'.
Enter new UNIX password:
Retype new UNIX password:
passwd: password updated successfully
Changing the user information for apollonio
Enter the new value, or press ENTER for the default
root@ip-xxx-xxx-xxx-xxx:~# addgroup apollonio
addgroup: The group `apollonio' already exists.
root@ip-xxx-xxx-xxx-xxx:~# delgroup apollonio
/usr/sbin/delgroup: `apollonio' still has `apollonio' as their primary group!
root@ip-xxx-xxx-xxx-xxx:~# deluser apollonio
Removing user `apollonio' ...
Warning: group `apollonio' has no more members.
Done.
root@ip-xxx-xxx-xxx-xxx:~# delgroup apollonio
The group `apollonio' does not exist.
root@ip-xxx-xxx-xxx-xxx:~#
```

Tested on debian:
```
         _,met$$$$$gg.           root@ip-xxx-xxx-xxx-xxx
      ,g$$$$$$$$$$$$$$$P.        OS: Debian 8.5 jessie
    ,g$$P""       """Y$$.".      Kernel: x86_64 Linux 3.16.0-4-amd64
   ,$$P'              `$$$.      Uptime: 265d 1h 20m
  ',$$P       ,ggs.     `$$b:    Packages: 705
  `d$$'     ,$P"'   .    $$$     Shell: bash 4.3.30
   $$P      d$'     ,    $$P     CPU: Intel Xeon CPU E5-2670 v2 @ 2.5GHz
   $$:      $$.   -    ,d$$'     RAM: 167MB / 998MB
   $$\;      Y$b._   _,d$P'
   Y$$.    `.`"Y$$$$P"'
   `$$b      "-.__
    `Y$$
     `Y$$.
       `$$b.
         `Y$$b.
            `"Y$b._
                `""""

```